pandas/io/excel/_odfreader.py

```diff
@@ -182,6 +183,20 @@ def _get_column_repeat(self, cell) -> int:
 
         return int(cell.attributes.get((TABLENS, "number-columns-repeated"), 1))
 
+    def _parse_odf_time(self, value: str) -> pd.Timestamp:
+        """
+        Helper function to convert ODF variant of ISO 8601 formatted duration
+        "PnYnMnDTnHnMnS" - see https://www.w3.org/TR/xmlschema-2/#duration
+        """
+        parts = re.match(r"^\s*PT\s*(\d+)\s*H\s*(\d+)\s*M\s*(\d+(\.\d+)?)\s*S$", value)
```

Comment:
This is going to be called many times, can you use a compiled regex.

Reply:
Good point, thanks! Fixed on 6ebe03a92d45802a84a504b31fd57a492e364ed5.